tests/bluetooth/iso/modules/iso_broadcast_sink.c

```diff
@@ -0,0 +1,529 @@
+			ret = gpio_pin_set_dt(&led, 1);
+		} else {
+			ret = gpio_pin_set_dt(&led, 0);
+		}
+		if (ret) {
+			LOG_ERR("Unable to set LED");
+		}
+	}
+}
+
+static void iso_connected(struct bt_iso_chan *chan)
+{
+	LOG_INF("ISO Channel %p connected", chan);
+	k_sem_give(&sem_big_sync);
+}
+
+static void iso_disconnected(struct bt_iso_chan *chan, uint8_t reason)
+{
+	LOG_INF("ISO Channel %p disconnected with reason 0x%02x", chan, reason);
+
+	if (reason != BT_HCI_ERR_OP_CANCELLED_BY_HOST) {
+		k_sem_give(&sem_big_sync_lost);
+	}
+}
+
+static struct bt_iso_chan_ops iso_ops = {
+	.recv = iso_recv,
+	.connected = iso_connected,
+	.disconnected = iso_disconnected,
+};
+
+static struct bt_iso_chan_io_qos iso_rx_qos[CONFIG_BIS_ISO_CHAN_COUNT_MAX];
+static struct bt_iso_chan_qos bis_iso_qos[CONFIG_BIS_ISO_CHAN_COUNT_MAX];
+static struct bt_iso_chan bis_iso_chan[CONFIG_BIS_ISO_CHAN_COUNT_MAX];
+static struct bt_iso_chan *bis[CONFIG_BIS_ISO_CHAN_COUNT_MAX];
+
+static struct bt_iso_big_sync_param big_sync_param = {
+	.bis_channels = bis,
+	.num_bis = 1,
+	.bis_bitfield = (BIT_MASK(1) << 1),
+	.mse = 1,
+	.sync_timeout = 100, /* in 10 ms units */
+};
+
+static void broadcaster_sink_trd(void)
+{
+	int ret;
+	struct bt_le_per_adv_sync_param sync_create_param;
+	struct bt_le_per_adv_sync *sync;
+	struct bt_iso_big *big;
+	uint32_t sem_retry_timeout_ms;
+
+	k_sem_take(&sem_running, K_FOREVER);
+
+	bt_le_scan_cb_register(&scan_callbacks);
+	bt_le_per_adv_sync_cb_register(&sync_callbacks);
+
+	while (1) {
+		per_adv_lost = false;
+
+		LOG_INF("Start scanning...");
+		ret = bt_le_scan_start(BT_LE_SCAN_CUSTOM, NULL);
+		if (ret) {
+			LOG_ERR("failed (ret %d)", ret);
+			return;
+		}
+
+		LOG_INF("Waiting for periodic advertising...");
+		per_adv_found = false;
+		ret = k_sem_take(&sem_per_adv, K_FOREVER);
+		if (ret) {
+			LOG_ERR("failed (ret %d)", ret);
+			return;
+		}
+
+		LOG_INF("Found periodic advertising.");
+
+		LOG_INF("Stop scanning...");
+		ret = bt_le_scan_stop();
+		if (ret) {
+			LOG_ERR("failed (ret %d)", ret);
+			return;
+		}
+
+		LOG_INF("Creating Periodic Advertising Sync...");
+		bt_addr_le_copy(&sync_create_param.addr, &per_addr);
+		sync_create_param.options = 0;
+		sync_create_param.sid = per_sid;
+		sync_create_param.skip = 0;
+		/* Multiple PA interval with retry count and convert to unit of 10 ms */
+		sync_create_param.timeout =
+			(per_interval_us * PA_RETRY_COUNT) / (10 * USEC_PER_MSEC);
+
+		sem_retry_timeout_ms = per_interval_us * PA_RETRY_COUNT;
+		ret = bt_le_per_adv_sync_create(&sync_create_param, &sync);
+		if (ret) {
+			LOG_ERR("failed (ret %d)", ret);
+			return;
+		}
+
+		LOG_INF("Waiting for periodic sync...");
+		ret = k_sem_take(&sem_per_sync, K_USEC(sem_retry_timeout_ms));
+		if (ret) {
+			LOG_ERR("failed (ret %d)", ret);
+
+			LOG_INF("Deleting Periodic Advertising Sync...");
+			ret = bt_le_per_adv_sync_delete(sync);
+			if (ret) {
+				LOG_ERR("failed (ret %d)", ret);
+				return;
+			}
+			continue;
```

Comment:
Still some places without newline after '}', but not critical if you merge it as it is